nxdrive/engine/engine.py

```diff
@@ -698,12 +698,16 @@ def cancel_upload(self, transfer_uid: int) -> None:
         upload = self.dao.get_dt_upload(uid=transfer_uid)
         if not upload:
             return
-        self.remote.cancel_batch(upload.batch)
-        self.dao.remove_transfer("upload", upload.path, is_direct_transfer=True)
-
         doc_pair = self.dao.get_state_from_local(upload.path)
         if not doc_pair:
             return
+        if upload.status is TransferStatus.ONGOING and doc_pair.processor:
+            upload.status = TransferStatus.CANCELLED
+            self.dao.set_transfer_status("upload", upload)
+            return
+        self.remote.cancel_batch(upload.batch)
```

Comment:
Same here. It was not clear for me and for history it would be good to add a comment.